### PR TITLE
tests: hide a HostDisk-specific function

### PR DESCRIPTION
**Pull request by dankenigsberg**
RandTmpDir() was used only to generate a path for HostDisk. I don't think that the tests that use it should exist as e2e test, but I do not address this here.

The function is named according to its purpose and placed in a more sensible package (`storage`)

```release-note
NONE
```

/sig code-quality

### CODE REVIEW

File: tests/storage/hostdisk.go [hunk 1]
```diff
@@ -44,3 +45,8 @@ func CreateDiskOnHost(diskPath string) *k8sv1.Pod {
 
 	return pod
 }
+
+func HostDiskDir() string {
```
Comment (EdDev) [inline tests/storage/hostdisk.go:49]: I think we have two items with this:

- The package this is defined in contains tests and not a library. Importing it into other places is not advised.
   I see it was already used this way, which is unfortunately.
   Any usage of `storage.*` should move the shared code to `libstorage` package.

- If a general temporary directory is needed, it is better to use something like:
```go
tmpDir = GinkgoT().TempDir()
```

It implicitly includes a teardown which removes the directory.

The open question is if `"/var/provision/kubevirt.io/tests"`path has any special meaning.
I do see some ref to it in regards to "mountinfo".

Perhaps we should handle for now the first issue and let the other be considered in the future.

**Review by EdDev — CHANGES_REQUESTED** [review]

File: tests/storage/hostdisk.go (hunk 1 above)
Comment (dankenigsberg) [reply tests/storage/hostdisk.go:49]: > Any usage of storage.* should move the shared code to libstorage package.

The code now lies is a grand junkyard. It is moved into a smaller junkyard, and as such I see this PR as a small improvement.
Would you elaborate on why you believe `storage` should not be important from other tests?

> If a general temporary directory is needed, it is better to use something like GinkgoT().TempDir()

Right, but unfortunately the test that use this function don't use a proper temporary directory; they create a directory on one of the worker nodes, which GinkgoT() cannot reach. The entire tests are problematic, as they take a lot of care about testing implementation details. I don't have the capacity or knowledge to understand and fix them, so in this PR I opt to contain their damage.

Comment (EdDev) [reply tests/storage/hostdisk.go:49]: > > Any usage of storage.* should move the shared code to libstorage package.
> 
> The code now lies is a grand junkyard. It is moved into a smaller junkyard, and as such I see this PR as a small improvement. Would you elaborate on why you believe `storage` should not be important from other tests?

It is not recommended to import test files, one should import packages that have no tests in them.
In this case, there is such a package: `libstorage`.

Comment (dankenigsberg) [reply tests/storage/hostdisk.go:49]: > It is not recommended to import test files, one should import packages that have no tests in them.

Would you educate me: why is this distinction between `storage` and `libstorage` important? I would have understood this if `libstorage` had been a general-purpose test-independent package, but that is not the case.

Comment (EdDev) [reply tests/storage/hostdisk.go:49]: `libstorage` should contain only helpers that support the e2e tests, no tests.
`storage` should contain tests and helpers that serves specific test files. The helpers should not be shared between test files.

It is also totally fine to move all helpers to `libstorage` even if they are not shared between test files.

Potentially, `libstorage` can be used by non storage tests.

This is the pattern/rules we try to enforce with `libnet` and `network`.

**Comment by dankenigsberg** [comment]
@alromeros is doing much of this better in https://github.com/kubevirt/kubevirt/pull/12289

**Pull request closed without merging**